### PR TITLE
hopefully fixes my 4gb of runtime #23058

### PR DESCRIPTION
**Pull request by Militaires**
wew

Closes #23058

### CODE REVIEW

**Comment by Militaires** [comment]
shit's fucked with the proc i know hold on

**Comment by KorPhaeron** [comment]
>hopefully

**Comment by Supermichael777** [comment]
WHAT DID THOU DO

File: code/modules/vehicles/speedbike.dm [hunk 1]
```diff
@@ -15,7 +15,6 @@
 	overlay = image("icons/obj/bike.dmi", overlay_state)
 	overlay.layer = ABOVE_MOB_LAYER
 	add_overlay(overlay)
-	riding_datum = new/datum/riding/space/speedbike
```
Comment (AnturK) [inline code/modules/vehicles/speedbike.dm]: Why this change ?

File: code/modules/vehicles/vehicle.dm [hunk 2]
```diff
@@ -28,8 +28,9 @@
 
 //BUCKLE HOOKS
 /obj/vehicle/unbuckle_mob(mob/living/buckled_mob,force = 0)
-	riding_datum.restore_position(buckled_mob)
-	. = ..()
+	if(riding_datum)
```
Comment (AnturK) [inline code/modules/vehicles/vehicle.dm]: Shouldn't these be mandatory for vehicles ? I suppose broken subtypes could use it.

File: code/modules/vehicles/speedbike.dm (hunk 1 above)
Comment (Militaires) [inline code/modules/vehicles/speedbike.dm]: because i changed it to declare a new datum only upon buckle_mob

Comment (AnturK) [inline code/modules/vehicles/speedbike.dm]: Oh right, forgot i even suggested that.

**Comment by Cyberboss** [comment]
Add `Closes #23058` to the body
